app/src/datasets/chorus_dt/places_resolution.py

```diff
@@ -84,30 +86,33 @@ def compute_distance(df, n):
     places["lat"] = places["lat"].astype(float)
     places["lon"] = places["lon"].astype(float)
     places_dict = places.set_index("place").to_dict()
+    return places, places_dict
+
+
+def main():
+    prestation_types = ["A", "AM", "AU", "T", "TC", "TCA", "TM", "TU"]
+    data = dataset.get_data("/data/raw/chorus-dt", prestation_types)
+    places = dataset.get_places(data)
+    places, places_dict = resolve_place(places)
 
     # when analyzing plane trips, doesn't work if
     # 'no stop' in trips columns. (ie. always since we programed
     # plane trips this way.)
-
-    for place_index in [0, 1, 2]:
-        trips["coords_place_%d_lat" % place_index] = trips[
-            "trip_place_%d" % place_index
-        ].apply(lambda x: None if x == 'No stop' else places_dict["lat"][x])
-        trips["coords_place_%d_lon" % place_index] = trips[
-            "trip_place_%d" % place_index
-        ].apply(lambda x: None if x == 'No stop' else places_dict["lon"][x])
-        trips["coords_place_%d" % place_index] = (
-            trips["coords_place_%d_lon" % place_index].astype(str)
-            + ";"
-            + trips["coords_place_%d_lat" % place_index].astype(str)
+    for column in ["lieu_depart", "lieu_arrivee", "lieu_etape"]:
+        data["%s_resolved" % column] = data[column].apply(lambda x: places_dict["resolved"][x])
+        data["%s_lat" % column] = data[column].apply(
+            lambda x: places_dict["lat"][x] if places_dict["resolved"][x] else np.NaN
+        )
+        data["%s_lon" % column] = data[column].apply(
+            lambda x: places_dict["lon"][x] if places_dict["resolved"][x] else np.NaN
```

Comment:
Probably out of this PR scope, but would be interested to know why we use a dictionary+apply instead of built-in pandas?

Reply:
You are right, all of this should be refactored, and even the code usage.
I might spend some time this afternoon to work on this, and use `pd.merge` instead of these horrible `apply`.
New issue here: https://github.com/dataforgoodfr/batch7_beges/issues/8